ami_backup.py

```diff
@@ -2,98 +2,49 @@
 #
 # @author Robert Kozora <bobby@kozora.me>
 #
-# This script will search for all instances having a tag with "Backup" or "backup"
-# on it. As soon as we have the instances list, we loop through each instance
-# and create an AMI of it. Also, it will look for a "Retention" tag key which
-# will be used as a retention policy number in days. If there is no tag with
-# that name, it will use a 7 days default value for each AMI.
+# "retention_days" is environment variable which will be used as a retention policy number in days. If there is no
+# environment variable with that name, it will use a 14 days default value for each AMI.
 #
-# After creating the AMI it creates a "DeleteOn" tag on the AMI indicating when
+# After creating the AMI it creates a "ami_delete_on" tag on the AMI indicating when
 # it will be deleted using the Retention value and another Lambda function 
 
+from __future__ import print_function
 import boto3
 import collections
 import datetime
 import sys
 import pprint
+import os
 
 ec = boto3.client('ec2')
-#image = ec.Image('id')
+ec2_instance_id = os.environ['instance_id']
+label_id = os.environ['label_id']
 
 def lambda_handler(event, context):
-    
-    reservations = ec.describe_instances(
-        Filters=[
-            {'Name': 'tag-key', 'Values': ['backup', 'Backup', 'Snapshot']},
+    try:
+        retention_days = os.environ['retention_days']
+    except:
```

Comment:
It is good practice to use named exceptions. In this way we know what exception happened and why we use it.
Let's use `except ValueError:`